pulser-core/pulser/backend/results.py

```diff
@@ -60,6 +60,10 @@ def _store(
                 f" at time {time}."
             )
         self._tagmap[observable.tag] = observable.uuid
+        try:
```

Comment:
what about
`assert _times == [] or _times[-1] < time`
instead of the try-catch block?